ami/trainers/i_jepa_latent_visualization_trainer.py

```diff
@@ -0,0 +1,217 @@
+import time
+from functools import partial
+from pathlib import Path
+from typing import Literal
+
+import torch
+import torch.nn.functional as F
+import torchvision.transforms.v2.functional as torchvisF
+import torchvision.utils
+from torch import Tensor
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader, Dataset
+from typing_extensions import override
+
+from ami.data.buffers.buffer_names import BufferNames
+from ami.data.buffers.random_data_buffer import RandomDataBuffer
+from ami.data.interfaces import ThreadSafeDataUser
+from ami.models.bool_mask_i_jepa import BoolMaskIJEPAEncoder
+from ami.models.i_jepa import IJEPAEncoder
+from ami.models.i_jepa_latent_visualization_decoder import (
+    IJEPALatentVisualizationDecoder,
+)
+from ami.models.model_names import ModelNames
+from ami.models.model_wrapper import ModelWrapper
+from ami.tensorboard_loggers import StepIntervalLogger
+
+from .base_trainer import BaseTrainer
+
+
+class IJEPALatentVisualizationTrainer(BaseTrainer):
+    """Trainer for IJEPA Decoder."""
+
+    @override
+    def __init__(
+        self,
+        partial_dataloader: partial[DataLoader[Tensor]],
+        partial_optimizer: partial[Optimizer],
+        device: torch.device,
+        logger: StepIntervalLogger,
+        encoder_name: Literal[ModelNames.I_JEPA_CONTEXT_ENCODER, ModelNames.I_JEPA_TARGET_ENCODER],
+        decoder_name: Literal[ModelNames.I_JEPA_CONTEXT_DECODER, ModelNames.I_JEPA_TARGET_DECODER],
```

Comment:
ここ`decoder_name`だけあればいい気がする
```suggestion
        decoder_name: Literal[ModelNames.I_JEPA_CONTEXT_DECODER, ModelNames.I_JEPA_TARGET_DECODER],
```

Reply:
確かに。